### PR TITLE
add function calcIDataTransTech

### PR DESCRIPTION
**Pull request by Fsioutas**
add function calcIDataTransTech

### CODE REVIEW

**Review by giannou — CHANGES_REQUESTED** [review]
adding an updated version of the `sets.gms` file from OPEN-PROM might be needed

File: R/calcIDataTransTech.R [hunk 1]
```diff
@@ -0,0 +1,153 @@
+#' calcIDataTransTech
+#'
+#' Use data to derive OPENPROM input parameter iDataTransTech
```
[the remark this answers is not in the document]
Comment (Fsioutas) [reply R/calcIDataTransTech.R]: ok

File: R/calcIDataTransTech.R [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  
+  a1 <- readSource("TechCosts", subtype = "Medium_cars")
+  a2 <- readSource("TechCosts", subtype = "Rail")
+  a3 <- readSource("TechCosts", subtype = "Aviation")
+  a4 <- readSource("TechCosts", subtype = "Inland_navigation")
+  a5 <- readSource("TechCosts", subtype = "HGVs>16t")
+  
+  q <- mbind(a1, a2, a3, a4, a5)
+  q <- as.quitte(q)
+  q$efficiency_value <- sub("_", ".", q$efficiency_value)
```
[the remark this answers is not in the document]
Comment (Fsioutas) [reply R/calcIDataTransTech.R]: ok

File: R/calcIDataTransTech.R [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  map <- toolGetMapping(name = "iDataTransTech-mapping.csv",
+                        type = "sectoral",
+                        where = "mappingfolder")
+  # load current OPENPROM set configuration
+  TRANSFINAL <- readSets(system.file(file.path("extdata", "sets.gms"), package = "mrprom"), "TRANSFINAL")
+  TRANSFINAL <- unlist(strsplit(TRANSFINAL[, 1], ","))
+  
+  TTECH <- readSets(system.file(file.path("extdata", "sets.gms"), package = "mrprom"), "TTECH")
+  TTECH <- unlist(strsplit(TTECH[, 1], ","))
+  years <- c(2015, 2020, 2030, 2040, 2050)
```
[the remark this answers is not in the document]
Comment (Fsioutas) [reply R/calcIDataTransTech.R]: ok

File: R/calcIDataTransTech.R [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  }
+  efficiency_value <- NULL
+  q <- mutate(q, mean_of_eff = mymedian(efficiency_value), .by = c("variable"))
+  
+  for (i in 1:nrow(map)) {
+    index11 <- which(x$Var2 == map[i, 2] & x$Var1 == map[i, 1] & x$Var3 != 2015)
+    index5 <- which(q$variable == map[i, 3])
+    index8 <- which(q$efficiency_value == q$mean_of_eff & q$variable == map[i, 3] & !(is.na(q$value)))
+    index9 <- which(q$variable == map[i, 3] & q$period == 2015 & !(is.na(q$value)))
+    index13 <- which(x$Var3 == 2015 & x$Var2 == map[i, 2] & x$Var1 == map[i, 1])
```
[the remark this answers is not in the document]
Comment (Fsioutas) [reply R/calcIDataTransTech.R]: ok

**Review by giannou — APPROVED** [review]

**Pull request merged**